### ISSUE TITLE
[imgui-sfml imguizmo implot milerius-sfml-imgui ogre] build failure

### ISSUE THREAD

**Issue by StarGate-One** [microsoft/vcpkg#35766]
### Operating system

Windows 11 10.0.22631.2861 x64 Enterprise

### Compiler

Visual Studio Community 2022 Version: [17.8.3]  VS Tools Version: [14.37.32822] 
Microsoft (R) C/C++ Optimizing Compiler Version [19.37.32826.1] for x64 

### Steps to reproduce the behavior

After commit: 70ce0e8d45d7acc016d1efa4e7c4255e3e86fa35
[[imgui] fix allegro feature #35729](https://github.com/microsoft/vcpkg/pull/35729)

```Shell
Vcpkg Version [2023-12-11-1c9ec1978a6b0c2b39c9e9554a96e3e275f7556e] 
Vcpkg Scripts Version [2023-12-18-48e95d1aa3fedd24234f1d436226d07fc2b4f45f] 

All failed for the following as all the ports depend on ports allegro and imgui.
> CMake Error at D:/vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package):
>  Could not find a package configuration file provided by
>  "unofficial-allegro5" with any of the following names:
>
>    unofficial-allegro5Config.cmake
>    unofficial-allegro5-config.cmake

Logs attached in `Additional Context` Section Below.

`.\vcpkg.exe install imgui-sfml imguizmo imgui[allegro5-binding,core,docking-experimental,dx10-binding,dx11-binding,dx12-binding,dx9-binding,freetype,glfw-binding,glut-binding,libigl-imgui,opengl3-binding,sdl2-binding,sdl2-renderer-binding,vulkan-binding,wchar32,win32-binding] implot milerius-sfml-imgui ogre[assimp,core,freeimage,openexr,overlay,strict,tools,zip,zziplib]`
```

### Failure logs

```
The following packages will be built and installed:
    imgui-sfml:x64-windows@2.6
    imguizmo:x64-windows@1.83#1
    implot:x64-windows@0.16
    milerius-sfml-imgui:x64-windows@1.1#6
    ogre[assimp,core,freeimage,openexr,overlay,strict,tools,zip,zziplib]:x64-windows@14.0.1#2

Installing 1/5 imgui-sfml:x64-windows@2.6...
Building imgui-sfml:x64-windows@2.6...
-- Note: imgui-sfml only supports static library linkage. Building static library.
-- Using cached eliasdaler-imgui-sfml-v2.6.tar.gz.
-- Extracting source D:/vcpkg/downloads/eliasdaler-imgui-sfml-v2.6.tar.gz
-- Applying patch 0001-fix_find_package.patch
-- Using source at D:/vcpkg/buildtrees/imgui-sfml/src/v2.6-1dc04eadfd.clean
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: D:/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe -v
    Working Directory: D:/vcpkg/buildtrees/imgui-sfml/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      D:\vcpkg\buildtrees\imgui-sfml\config-x64-windows-dbg-CMakeCache.txt.log
      D:\vcpkg\buildtrees\imgui-sfml\config-x64-windows-rel-CMakeCache.txt.log
      D:\vcpkg\buildtrees\imgui-sfml\config-x64-windows-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:252 (vcpkg_execute_required_process)
  ports/imgui-sfml/portfile.cmake:13 (vcpkg_cmake_configure)
  scripts/ports.cmake:170 (include)


error: building imgui-sfml:x64-windows failed with: BUILD_FAILED
Elapsed time to handle imgui-sfml:x64-windows: 2.1 s
Installing 2/5 imguizmo:x64-windows@1.83#1...
Building imguizmo:x64-windows@1.83#1...
-- Note: imguizmo only supports static library linkage. Building static library.
-- Using cached CedricGuillemet-ImGuizmo-1.83.tar.gz.
-- Extracting source D:/vcpkg/downloads/CedricGuillemet-ImGuizmo-1.83.tar.gz
-- Applying patch fix-build.patch
-- Using source at D:/vcpkg/buildtrees/imguizmo/src/1.83-92328c0b93.clean
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: D:/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe -v
    Working Directory: D:/vcpkg/buildtrees/imguizmo/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      D:\vcpkg\buildtrees\imguizmo\config-x64-windows-dbg-CMakeCache.txt.log
      D:\vcpkg\buildtrees\imguizmo\config-x64-windows-rel-CMakeCache.txt.log
      D:\vcpkg\buildtrees\imguizmo\config-x64-windows-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:252 (vcpkg_execute_required_process)
  ports/imguizmo/portfile.cmake:15 (vcpkg_cmake_configure)
  scripts/ports.cmake:170 (include)


error: building imguizmo:x64-windows failed with: BUILD_FAILED
Elapsed time to handle imguizmo:x64-windows: 1.5 s
Installing 3/5 implot:x64-windows@0.16...
Building implot:x64-windows@0.16...
-- Note: implot only supports static library linkage. Building static library.
-- Using cached epezent-implot-v0.16.tar.gz.
-- Extracting source D:/vcpkg/downloads/epezent-implot-v0.16.tar.gz
-- Using source at D:/vcpkg/buildtrees/implot/src/v0.16-45ccd72192.clean
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: D:/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe -v
    Working Directory: D:/vcpkg/buildtrees/implot/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      D:\vcpkg\buildtrees\implot\config-x64-windows-dbg-CMakeCache.txt.log
      D:\vcpkg\buildtrees\implot\config-x64-windows-rel-CMakeCache.txt.log
      D:\vcpkg\buildtrees\implot\config-x64-windows-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:252 (vcpkg_execute_required_process)
  ports/implot/portfile.cmake:13 (vcpkg_cmake_configure)
  scripts/ports.cmake:170 (include)


error: building implot:x64-windows failed with: BUILD_FAILED
Elapsed time to handle implot:x64-windows: 1.1 s
Installing 4/5 milerius-sfml-imgui:x64-windows@1.1#6...
Building milerius-sfml-imgui:x64-windows@1.1#6...
-- Using cached Milerius-imgui-sfml-shared-1.1.tar.gz.
-- Extracting source D:/vcpkg/downloads/Milerius-imgui-sfml-shared-1.1.tar.gz
-- Applying patch FixFindPackageIssue.patch
-- Applying patch cpp11.patch
-- Applying patch fix-imgui-dependency.patch
-- Using source at D:/vcpkg/buildtrees/milerius-sfml-imgui/src/1.1-2cfd091fae.clean
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: D:/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe -v
    Working Directory: D:/vcpkg/buildtrees/milerius-sfml-imgui/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      D:\vcpkg\buildtrees\milerius-sfml-imgui\config-x64-windows-dbg-CMakeCache.txt.log
      D:\vcpkg\buildtrees\milerius-sfml-imgui\config-x64-windows-rel-CMakeCache.txt.log
      D:\vcpkg\buildtrees\milerius-sfml-imgui\config-x64-windows-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:252 (vcpkg_execute_required_process)
  ports/milerius-sfml-imgui/portfile.cmake:15 (vcpkg_cmake_configure)
  scripts/ports.cmake:170 (include)


error: building milerius-sfml-imgui:x64-windows failed with: BUILD_FAILED
Elapsed time to handle milerius-sfml-imgui:x64-windows: 1.7 s
Installing 5/5 ogre[assimp,core,freeimage,openexr,overlay,strict,tools,zip,zziplib]:x64-windows@14.0.1#2...
Building ogre[assimp,core,freeimage,openexr,overlay,strict,tools,zip,zziplib]:x64-windows@14.0.1#2...
-- Using cached OGRECave-ogre-v14.0.1.tar.gz.
-- Extracting source D:/vcpkg/downloads/OGRECave-ogre-v14.0.1.tar.gz
-- Applying patch fix-dependencies.patch
-- Applying patch cfg-rel-paths.patch
-- Applying patch swig-python-polyfill.patch
-- Applying patch pkgconfig.patch
-- Applying patch same-install-rules-all-platforms.patch
-- Using source at D:/vcpkg/buildtrees/ogre/src/v14.0.1-e1473b508d.clean
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: D:/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe -v
    Working Directory: D:/vcpkg/buildtrees/ogre/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      D:\vcpkg\buildtrees\ogre\config-x64-windows-dbg-CMakeCache.txt.log
      D:\vcpkg\buildtrees\ogre\config-x64-windows-rel-CMakeCache.txt.log
      D:\vcpkg\buildtrees\ogre\config-x64-windows-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:252 (vcpkg_execute_required_process)
  ports/ogre/portfile.cmake:71 (vcpkg_cmake_configure)
  scripts/ports.cmake:170 (include)


error: building ogre:x64-windows failed with: BUILD_FAILED
Elapsed time to handle ogre:x64-windows: 5.9 s
Total install time: 12 s

RESULTS
  imgui-sfml:x64-windows: BUILD_FAILED: 2.1 s
  imguizmo:x64-windows: BUILD_FAILED: 1.5 s
  implot:x64-windows: BUILD_FAILED: 1.1 s
  milerius-sfml-imgui:x64-windows: BUILD_FAILED: 1.7 s
  ogre:x64-windows: BUILD_FAILED: 5.9 s

SUMMARY FOR x64-windows
  BUILD_FAILED: 5
```

### Additional context

Attached logs:
[console.log](https://github.com/microsoft/vcpkg/files/13710650/console.log)
[imgui-sfml_config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/13710651/imgui-sfml_config-x64-windows-out.log)
[imguizmo_config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/13710652/imguizmo_config-x64-windows-out.log)
[implot_config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/13710653/implot_config-x64-windows-out.log)
[milerius-sfml-imgui_config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/13710654/milerius-sfml-imgui_config-x64-windows-out.log)
[ogre_config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/13710655/ogre_config-x64-windows-out.log)

**Comment by MonicaLiu0311** [comment]
> After commit: [70ce0e8](https://github.com/microsoft/vcpkg/commit/70ce0e8d45d7acc016d1efa4e7c4255e3e86fa35)
> [[imgui] fix allegro feature #35729](https://github.com/microsoft/vcpkg/pull/35729)

Two weeks ago, #35495 updated `allegro5` to 5.2.9.0. The new version officially exported the `config` series files, so the outdated patch of vcpkg was removed (the patch exported the `config` file prefixed with `unofficial-`), which caused downstream usage No longer applicable.

And #35729 is to accommodate the changes in `allegro5`, not the issue it caused.

I'll fix this right away, thanks for reporting it.

**Comment by StarGate-One** [comment]
Thank you all very much!!!
